### ISSUE TITLE
Interfaces should be prefixed with I*

### ISSUE THREAD

**Issue by wictorwilen** [OfficeDev/msteams-ui-components#147]
I'd recommend all interfaces to be refactored so that they start with an `I`. For instance:

`TeamsComponentProps` should be `ITeamsComponentProps` and `TeamsComponentState` to `ITeamsComponentState`

That way we follow good programming practices, and also align across how other MIcrosoft packages are designed (SPFx, Office UI Fabric React etc)

**Comment by MattSFT** [comment]
Agreed. Feel free to submit a pull request :).

**Comment by wictorwilen** [comment]
I'm on it!

**Comment by wictorwilen** [comment]
Here it is #149

**Comment by MattSFT** [comment]
published your changes as version 0.5.0 for styles-core and components-react and 0.3.0 for the icons packages

**Comment by wictorwilen** [comment]
Awesome!

**Comment by greenpau** [comment]
@wictorwilen , @MattSFT , I no longer see `ITeamsComponentState` in `msteams-ui-components-react`, where did it move?

**Comment by wictorwilen** [comment]
@greenpau - I'd recommend not using this library as there is no future in this. For Teams (and all M365) UX components please look at this announcement: https://developer.microsoft.com/en-us/office/blogs/ui-fabric-is-evolving-into-fluent-ui/

**Comment by greenpau** [comment]
> @greenpau - I'd recommend not using this library as there is no future in this. For Teams (and all M365) UX components please look at this announcement: https://developer.microsoft.com/en-us/office/blogs/ui-fabric-is-evolving-into-fluent-ui/

@wictorwilen , thank you for letting me know! Originally, I generated an app with `yo teams`. As I was learning the flow, it became clear that some components are no longer available with the latest UI. It was a bit confusing.

At the moment, I used `npm init uifabric` to bootstrap an app. It does not have MS Teams integration, but it would work to learn the components and the state machine.
